app/services/doit/post_to_doit.rb

```diff
@@ -19,7 +19,8 @@ def call
       options[:headers] = {'X-API-Key' => DOIT_AUTH_TOKEN,
                            'Content-Type' => 'application/json'}
       options[:body] = build_request_body
-      http_adapter.post("#{ENV['DOIT_HOST']}#{doit_volonteer_op_resource}", options)
+      doit_res = http_adapter.post(doit_volonteer_op_resource, options)
+      JSON.parse(doit_res.body)['data']['opportunity']['id']
```

Comment:
@tansaku the `call` was originally a command method, but to store the generated it now parse and return the id, do you consider it as a violation of command-query separation? I tried to make the return optional via blocks but I didn't manage to achieve it.

Reply:
ah hmm, I'll have to look more closely at this ...